### PR TITLE
Remove wrong code snippets from error message when in import-ast mode

### PR DESCRIPTION
**Pull request by matheusaaguiar**
This should fix the issue #12168.

### CODE REVIEW

File: liblangutil/SourceReferenceFormatter.cpp [hunk 1]
```diff
@@ -124,6 +124,9 @@ void SourceReferenceFormatter::printSourceLocation(SourceReference const& _ref)
 
 	string_view text = _ref.text;
 
+	if(m_charStreamProvider.charStream(_ref.sourceName).isImportedFromAST())
```
Comment (stackenbotten) [inline liblangutil/SourceReferenceFormatter.cpp]: Coding style error

File: libsolidity/interface/CompilerStack.cpp [hunk 2]
```diff
@@ -1286,7 +1286,7 @@ void CompilerStack::assemble(
 			"Contract code size is "s +
 			to_string(compiledContract.runtimeObject.bytecode.size()) +
 			" bytes and exceeds 24576 bytes (a limit introduced in Spurious Dragon). "
-			"This contract may not be deployable on Mainnet. "
```
Comment (Marenz) [inline libsolidity/interface/CompilerStack.cpp]: Bad rebase?

**Review by Marenz — CHANGES_REQUESTED** [review]

File: liblangutil/SourceReferenceFormatter.cpp [hunk 3]
```diff
@@ -124,6 +124,9 @@ void SourceReferenceFormatter::printSourceLocation(SourceReference const& _ref)
 
 	string_view text = _ref.text;
 
+	if (m_charStreamProvider.charStream(_ref.sourceName).isImportedFromAST())
+		return;
```
Comment (Marenz) [inline liblangutil/SourceReferenceFormatter.cpp:128]: Could we have at least one test that shows the changes?

File: libsolidity/interface/CompilerStack.cpp (hunk 2 above)
Comment (matheusaaguiar) [reply libsolidity/interface/CompilerStack.cpp]: Yeah, maybe, not sure what happened here ...

**Comment by matheusaaguiar** [comment]
Replicating the same test used in issue [12168](https://github.com/ethereum/solidity/issues/12168):
```
echo 'contract C { function foo(uint x) public { require(x > 0); } }' | \
    solc --combined-json ast - | \
    solc --import-ast -
```
Output of first `solc` command:

```
Warning: Function state mutability can be restricted to pure
 --> <stdin>:1:14:
  |
1 | contract C { function foo(uint x) public { require(x > 0); } }
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```
**Output of second `solc` command (Affected by the current PR):**
```
Warning: Function state mutability can be restricted to pure
 --> <stdin>:1:14:
```

**Comment by matheusaaguiar** [comment]
Looking at it now, i think would be good to include a informative message letting the user know that the function code snippet cannot be shown because the source was imported from json.
edit: nevermind. `--import-ast` is rarely used and the user probably will figure this out.

**Review by Marenz — APPROVED** [review]

**Pull request merged**